### PR TITLE
Download parameters saving

### PR DESCRIPTION
**Pull request by guynir42**
Save the parameters that were used to download the raw data. 
Then allow the user to check and replace any files that were generated with a different set of parameters.

### CODE REVIEW

**Comment by guynir42** [comment]
@felixzhu555 let me know if any parts of the new code are not clear, we can try to make it more readable or add comments where needed.

File: src/source.py [hunk 1]
```diff
@@ -312,6 +312,72 @@ def get_raw_data(self, obs, data_type=None, session=None):
...
+            for data in getattr(self, f"raw_{data_type}")
+            if data.observatory == obs
+        ]
+        if len(raw_data) == 0:
+            return False
+        elif len(raw_data) == 1:
+            raw_data = raw_data[0]
+            getattr(self, f"raw_{data_type}").remove(raw_data)
+            if session is not None:
+                session.delete(raw_data)
```
Comment (felixzhu555) [inline src/source.py:374]: should the elif block end with `return True`?

**Review by felixzhu555 — APPROVED** [review]
Documentation makes sense

File: src/source.py (hunk 1 above)
Comment (guynir42) [reply src/source.py:374]: Yes, you are right!

**Pull request merged**